worker/multiwatcher/worker.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package multiwatcher
+
+import (
+	"sync"
+
+	"github.com/juju/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/juju/worker.v1"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/core/multiwatcher"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/watcher"
+)
+
+// Config is an argument struct used to create a Worker.
+type Config struct {
+	Logger               Logger
+	Backing              state.AllWatcherBacking
+	PrometheusRegisterer prometheus.Registerer
+	Cleanup              func()
+}
+
+// Validate validates the worker configuration.
+func (config Config) Validate() error {
+	if config.Logger == nil {
+		return errors.NotValidf("missing Logger")
+	}
+	if config.Backing == nil {
+		return errors.NotValidf("missing Backing")
+	}
+	if config.PrometheusRegisterer == nil {
+		return errors.NotValidf("missing PrometheusRegisterer")
+	}
+	return nil
+}
+
+// Worker runs the primary goroutine for managing the multiwatchers.
+type Worker struct {
+	config Config
+
+	tomb    tomb.Tomb
+	metrics *Collector
+
+	// store holds information about all known entities.
+	store multiwatcher.Store
+
+	// request receives requests from Multiwatcher clients.
+	request chan *request
+
+	// Each entry in the waiting map holds a linked list of Next requests
+	// outstanding for the associated params.
+	waiting map[*Watcher]*request
+
+	mu           sync.Mutex
+	watchers     []*Watcher
+	restartCount int
+	// remember the last five errors that caused us to restart the internal loop
+	errors []error
+}
+
+// request holds a message from the Multiwatcher to the
+// storeManager for some changes. The request will be
+// replied to when some changes are available.
+type request struct {
+	// w holds the Multiwatcher that has originated the request.
+	watcher *Watcher
+
+	// reply receives a message when deltas are ready.  If reply is
+	// nil, the Multiwatcher will be stopped.  If the reply is true,
+	// the request has been processed; if false, the Multiwatcher
+	// has been stopped,
+	reply chan bool
+
+	// noChanges receives a message when the manager checks for changes
+	// and there are none.
+	noChanges chan struct{}
+
+	// On reply, changes will hold changes that have occurred since
```

Comment:
reform comment to start with "changes"

Reply:
I had figured since it was a internal member, and lint didn't complain it wasn't a big deal. Will update though.